solr/core/src/java/org/apache/solr/client/solrj/embedded/EmbeddedSolrServer.java

```diff
@@ -71,6 +71,7 @@
   protected final String coreName;
   private final SolrRequestParsers _parser;
   private final RequestWriterSupplier supplier;
+  private boolean containerIsLocal = false;
```

Comment:
Can this be final?

Reply:
I tried to do that, but it gets awkward because there's 5 constructors and one calls another which calls another. I'd have to refactor all of them to call a private constructor instead, to avoid changing the external constructor parameters. Do you think I should I do that?

Reply:
I think we can pass null from the other constructors and in the final constructor if the cc is still null we create one and set local to true?

Reply:
That would be good, but there's no context available to create a cc because the original parameters (e.g. NodeConfig or solrHome Path) aren't in scope then.

I'm not that worried about this boolean flag because it's private to the class so we know where it's set.